miden-tx/tests/integration/scripts/p2id.rs

```diff
@@ -7,23 +9,43 @@ use miden_objects::{
             ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN,
             ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN_2, ACCOUNT_ID_SENDER,
         },
-        Account, AccountId,
+        Account, AccountId, AccountType, SlotItem,
     },
     assembly::ProgramAst,
     assets::{Asset, AssetVault, FungibleAsset},
     crypto::rand::RpoRandomCoin,
     notes::{NoteScript, NoteType},
-    testing::{account_code::DEFAULT_AUTH_SCRIPT, notes::DEFAULT_NOTE_CODE},
+    testing::{
+        account::AccountBuilder, account_code::DEFAULT_AUTH_SCRIPT, notes::DEFAULT_NOTE_CODE,
+    },
     transaction::TransactionArgs,
-    Felt,
+    Felt, FieldElement,
 };
-use miden_tx::{testing::TransactionContextBuilder, TransactionExecutor};
+use miden_tx::{auth::BasicAuthenticator, testing::TransactionContextBuilder, TransactionExecutor};
+use rand::{rngs::StdRng, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+use vm_processor::Word;
 
 use crate::{
     get_account_with_default_account_code, get_new_pk_and_authenticator,
     prove_and_verify_transaction,
 };
 
+fn create_new_account() -> (Account, Word, Rc<BasicAuthenticator<StdRng>>) {
```

Comment:
nit: I'd probably move this to the end of the file into something like a `HELPER FUNCTIONS` section.